plugins/kafka/builtins/plugin.cpp

```diff
@@ -0,0 +1,97 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "kafka/operator.hpp"
+
+#include <tenzir/tql2/plugin.hpp>
+
+namespace tenzir::plugins::kafka {
+namespace {
+
+class load_plugin final
+  : public virtual operator_plugin2<loader_adapter<kafka_loader>> {
+public:
+  auto initialize(const record& config,
+                  const record& /* global_config */) -> caf::error override {
+    config_ = config;
+    if (!config_.contains("bootstrap.servers")) {
+      config_["bootstrap.servers"] = "localhost";
+    }
+    if (!config_.contains("client.id")) {
+      config_["client.id"] = "tenzir";
+    }
+    return caf::none;
+  }
+
+  auto
+  make(invocation inv, session ctx) const -> failure_or<operator_ptr> override {
+    auto args = loader_args{};
+    TRY(argument_parser2::operator_(name())
+          .add("topic", args.topic)
+          .add("count", args.count)
+          .add("exit", args.exit)
+          .add("offset", args.offset)
```

Comment:
I'm not sure whether `offset` should be a string here. There are some string constants like "beginning", "end" and "stored", but it can also accept a number. Your version doesn't allow `offset=42` and requires `offset="42"` instead.

Reply:
The offset is the same as in TQL1. What do you suggest doing with this? Taking both?

Reply:
Agreeing that we want either a string or number. How hard is it to patch support for this into the argument parser?

Reply:
Supporting this in the argument parser seems to require some thought. I've implemented this using `const_eval` for now. We can come back to this later.